watchtower-plugin/src/main.rs

```diff
@@ -133,10 +135,15 @@ async fn get_registration_receipt(
     plugin: Plugin<Arc<Mutex<WTClient>>>,
     v: serde_json::Value,
 ) -> Result<serde_json::Value, Error> {
-    let tower_id = TowerId::try_from(v).map_err(|x| anyhow!(x))?;
+    let params = GetRegistrationReceiptParams::try_from(v).map_err(|x| anyhow!(x))?;
+    let tower_id = params.tower_id;
+    let subscription_start = params.subscription_start;
+    let subscription_expiry = params.subscription_expiry;
     let state = plugin.state().lock().unwrap();
 
-    if let Some(response) = state.get_registration_receipt(tower_id) {
+    if let Some(response) =
+        state.get_registration_receipt(tower_id, subscription_start, subscription_expiry)
+    {
         Ok(json!(response))
     } else {
         Err(anyhow!(
```

Comment:
Notice this is not the right error for every case now. It used to be the case because if no receipt could be pulled from the database it meant the tower was unknown. Now, if params have been passed, it could be the case that you could not find a receipt for the specified boundary. This needs to be covered. I'd suggest you check whether the tower can be found in memory if the query errors, and the return the proper error depending on that outcome.